crates/api_common/src/custom_emoji.rs

```diff
@@ -0,0 +1,41 @@
+use crate::sensitive::Sensitive;
+use lemmy_db_views::structs::CustomEmojiView;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct CreateCustomEmoji {
+  pub category: String,
+  pub shortcode: String,
+  pub image_url: String,
+  pub alt_text: String,
+  pub keywords: String,
+  pub auth: Sensitive<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct EditCustomEmoji {
+  pub id: i32,
+  pub category: String,
+  pub shortcode: String,
+  pub image_url: String,
+  pub alt_text: String,
+  pub keywords: String,
```

Comment:
Since this is an optional vector, keywords should probably be similar to taglines, IE  `Option<Vec<String>>`. 

Make sure the CreateCustomEmoji.keywords field matches this one.

Reply:
I keep keywords as a string (of space-separated words) because handling it on the frontend with the gui is a lot more complicated in the current view, where all the keywords lie in a single text input field. My original implementation had the vec, but I couldn't get the data field to work properly with binding it to inferno input element.

Reply:
You'll have to show me the code where thats difficult, because it should be trivial to transform that field to whatever format emoji-mart needs before binding it.